pkg/grpc/client.go

```diff
@@ -0,0 +1,76 @@
+package grpc
+
+// gRPC client implementation to be able to connect to the gRPC server and perform the following operations:
+// - Start a remote JupyterLab server
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/cli/cli/v2/pkg/grpc/jupyter"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/metadata"
+)
+
+const (
+	requestTimeout = 30 * time.Second
+)
+
+type GrpcClient struct {
+	conn          *grpc.ClientConn
+	token         string
+	jupyterClient jupyter.JupyterServerHostClient
+}
+
+func New() *GrpcClient {
+	return &GrpcClient{}
+}
+
+// Connects to the gRPC server on the given port
+func (g *GrpcClient) Connect(ctx context.Context, port int, token string) error {
+	// Attempt to connect to the given port
+	conn, err := grpc.Dial(fmt.Sprintf("localhost:%d", port), grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithBlock())
+
+	if err != nil {
+		return fmt.Errorf("Failed to connect to the internal server on port %d", port)
```

Comment:
```suggestion
		return fmt.Errorf("failed to connect to the internal server on port %d: %w", port, err)
```

Underlying error should be wrapped and returned

Reply:
Are the lower case messages a part of the standard golang conventions? It seemed a bit inconsistent throughout the code so I wasn't sure.

Reply:
https://github.com/golang/go/wiki/CodeReviewComments#error-strings

They should be lowercase and have no punctuation. If you find instances that aren't, they should probably be updated (although I mostly only speak for the codespaces code, @mislav & team may have other opinions)

Reply:
Got it, thanks!